src/graph/optimizer/OptGroup.cpp

```diff
@@ -81,8 +81,12 @@ Status OptGroup::explore(const OptRule *rule) {
     NG_RETURN_IF_ERROR(resStatus);
     auto result = std::move(resStatus).value();
     if (result.eraseAll) {
-      for (auto gnode : groupNodes_) {
-        gnode->node()->releaseSymbols();
+      matched.eraseAll();  // erase current mached sub-plan
```

Comment:
Different OptNodes could share the same OptGroup as their dependencies. If one OptNode should be removed, we could not delete the OptGroup depended by it directly.

Reply:
As I known, the replaced sub-plan which matched is invisible again in the whole plan. Delete the information of them in symbol table is unsafe?

Reply:
Not the sub-plan, only the OptNode in the same OptGroup will be removed. The dependent OptGroup of the OptNode maybe be shared with other OptNodes in different OptGroups.

Reply:
But current implementation contains error. Optimizer will release the symbols(readBy, writtenBy) of replaced nodes, but now just release the top-layer nodes instead of all matched nodes. So some deprecated symbols(readBy, writtenBy) will keep alive in symbol table, which will prevent some valid optimization.